lib/cloud_controller/upload_buildpack.rb

```diff
@@ -47,8 +50,31 @@ def upload_buildpack(buildpack, bits_file_path, new_filename)
       true
     end
 
+    def extract_stack_from_buildpack(bits_file_path)
```

Comment:
This feels like a separate concern from uploading the buildpack. Since it is called by buildpack_installer and a private method in upload_buildpack, I would either move it or break it into its own class.

Reply:
Ok sure.